common/achievement/bucket.ts

```diff
@@ -12,17 +12,19 @@ export const bucketCreatorDefs: BucketCreatorDefs = {
         },
     },
     by_lecture_start: {
-        function: async (relation: string): Promise<GenericBucketConfig<TimeBucket>> => {
+        function: async (relation): Promise<GenericBucketConfig<TimeBucket>> => {
+            const [type] = getRelationTypeAndId(relation);
+
             if (!relation) {
                 return { bucketKind: 'time', buckets: [] };
             }
-            const context = await getRelationContext(relation);
-            if (!context[relation].lecture) {
+            const context = await getBucketContext(relation);
```

Comment:
this would complicate things, since we need to have the context before we know, what bucket kind we are searching for. Inside the function this is easy since we have the time defined by the function itself. I'd prefer not to force a way around this.

Reply:
Oh, looks like I made the same comment twice 😅 
https://github.com/corona-school/backend/pull/913#discussion_r1422906378

I don't understand the problem. Do we really need to know the type of bucket before gathering the context?